src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HandshakeProtocol.cs

```diff
@@ -101,7 +103,8 @@ public static bool TryParseRequestMessage(ReadOnlySequence<byte> buffer, out Han
                 var token = JToken.ReadFrom(reader);
                 var handshakeJObject = JsonUtils.GetObject(token);
                 var protocol = JsonUtils.GetRequiredProperty<string>(handshakeJObject, ProtocolPropertyName);
-                requestMessage = new HandshakeRequestMessage(protocol);
+                var protocolVersion = JsonUtils.GetRequiredProperty<int>(handshakeJObject, ProtocolVersionName);
```

Comment:
Check that version isn't negative/zero? Although I guess that will fail when checked against a protocol if the min version is 1. Up to you.